### DISCUSSION TITLE
Cleanup RoutingProfile

### DISCUSSION THREAD

**Discussion by sfendrich** [GIScience/openrouteservice#1987] [Refactoring]
### Problem description

RoutingProfile is a quite messy class and should be cleaned up.

### Proposed solution

- [x] Move out unrelated members (#1521)
- [x] Move compute and build methods into appropriate places (#1951 #1958 )
  - [x] Matrix (4 methods)
  - [x] Isochrones (2 methods)
  - [x] Route
  - [x] RoundTrip
  - [x] Snap (https://github.com/GIScience/HeiGIT-Github-hidden-Issues/issues/64)
  - [x] Export
- [ ] Refactor interface between ORS profiles and GH profiles

**Comment by MichaelsJP** [comment]
Partially fixed by GIScience/openrouteservice#1521
